packages/data/src/components/use-select/test/index.js

```diff
@@ -434,6 +434,67 @@ describe( 'useSelect', () => {
 			);
 		} );
 
+		it( 'captures state changes before the subscription is set', () => {
+			registry.registerStore( 'store-1', counterStore );
+			registry.registerStore( 'store-2', counterStore );
+			registry.registerStore( 'store-3', counterStore );
+
+			let childShouldDispatch, setChildShouldDispatch;
+			const selectCount1AndDep = jest.fn();
+
+			class ChildComponent extends Component {
+				componentDidUpdate( prevProps ) {
+					if (
+						this.props.childShouldDispatch &&
+						this.props.childShouldDispatch !==
+							prevProps.childShouldDispatch
+					) {
+						registry.dispatch( 'store-1' ).increment();
+					}
+				}
+
+				render() {
+					return null;
+				}
+			}
+
+			const TestComponent = jest.fn( () => {
+				[ childShouldDispatch, setChildShouldDispatch ] =
```

Comment:
Instead of storing a reference to `setChildShouldDispatch` and then calling it, we could achieve the same by rendering a button and triggering an event:
```js
<button onClick={ () => setChildShouldDispatch( true ) }>trigger</button>

fireEvent.click( rendered.getbyText( 'trigger' ) );
```
That's more in the spirit of React unit tests, working only with the DOM UI that any other user can also use, without accessing internals.